src/main/java/org/folio/circulation/domain/OverdueFineCalculatorService.java

```diff
@@ -0,0 +1,201 @@
+package org.folio.circulation.domain;
+
+import org.folio.circulation.domain.policy.OverdueFinePolicy.OverdueFineInterval;
+import org.folio.circulation.domain.policy.OverdueFinePolicyRepository;
+import org.folio.circulation.domain.representations.FeeFineRepresentation;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.ItemRepository;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.ResultBinding;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.support.Result.of;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.folio.circulation.support.ResultBinding.mapResult;
+
+public class OverdueFineCalculatorService {
+  private static class CalculationParameters {
+    Loan loan;
+    Item item;
+    FeeFineOwner feeFineOwner;
+    FeeFine feeFine;
+
+    CalculationParameters(Loan loan, Item item, FeeFineOwner feeFineOwner, FeeFine feeFine) {
+      this.loan = loan;
+      this.item = item;
+      this.feeFineOwner = feeFineOwner;
+      this.feeFine = feeFine;
+    }
+
+    CalculationParameters withItem(Item item) {
+      return new CalculationParameters(this.loan, item, this.feeFineOwner, this.feeFine);
+    }
+
+    CalculationParameters withOwner(FeeFineOwner feeFineOwner) {
+      return new CalculationParameters(this.loan, this.item, feeFineOwner, this.feeFine);
+    }
+
+    CalculationParameters withFeeFine(FeeFine feeFine) {
+      return new CalculationParameters(this.loan, this.item, this.feeFineOwner, feeFine);
+    }
+  }
+
+  private static final Map<OverdueFineInterval, Integer> MINUTES_IN_INTERVAL =
+    new EnumMap<>(OverdueFineInterval.class);
+  static {
+    MINUTES_IN_INTERVAL.put(OverdueFineInterval.MINUTE, 1);
+    MINUTES_IN_INTERVAL.put(OverdueFineInterval.HOUR, 60);
+    MINUTES_IN_INTERVAL.put(OverdueFineInterval.DAY, 1440);
+    MINUTES_IN_INTERVAL.put(OverdueFineInterval.WEEK, 10080);
+    MINUTES_IN_INTERVAL.put(OverdueFineInterval.MONTH, 44640);
+    MINUTES_IN_INTERVAL.put(OverdueFineInterval.YEAR, 525600);
+  }
```

Comment:
Given that a special class has been introduced for overdue fine intervals, could this map be moved to that class and a method used to do this conversion?

Reply:
Moved to OverdueFineInterval